source/aspnetintegration.rst

```diff
@@ -263,3 +266,210 @@ A typical solution to this problem is to split up the class into multiple smalle
+    }
+
+
+.. container:: Note
+
+    **NOTE**: The `documentation for hosted services <https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services/>`_ says to register these services with the services.AddHostedService<T> extension method but that extension method doesn't give a way to provide either an instance or a factory.  Internally, the AddHostedService<T> extension method just calls the AddTransient<IHostedService, THostedService>() method. Later on, all of the IHostedService implementations are added as a dependency to HostedServiceExecuter which itself is added as a singleton. So, adding the hosted service implementations as singletons to the ASP.NET Core configuration system is an equivalent (and much clearer) way to add hosted services to the ASP.NET Core configuration system.
+
+The approach that will be laid out next is the preferred approach and it involves creating an adapter implementation that gets registered with the ASP.NET Core configuration system and forwards the calls to Simple Injector with the application-specific abstraction.
+
+The main idea with this approach is to have an application specific abstraction or abstractions which define the functionality to be performed in the IHostedService.  As an example, let's call this application-specific abstraction "IMyJob".  The implementation of IHostedService then becomes a simple adapter which can be used by ASP.NET Core to run application-specific dependencies in a hosted service similar to the way that the IControllerActivator abstraction implemented by SimpleInjectorControllerActivator provides an adapter for ASP.NET Core to be able to create Controllers with dependencies resolved by the Container.
+
+Given that, the IHostedService adapter might look like this:
+
+.. code-block:: c#
+
+    public class SimpleInjectorJobProcessorHostedService : IHostedService, IDisposable
+    {
+        private readonly Container _container;
+        private Timer _timer;
+
+        //Depending on how this adapter gets registered with the ASP.NET Core configuration system, this
+        //constructor may be called before all of the registrations have been added to the container.
+        public SimpleInjectorJobProcessorHostedService(Container container) => _container = container;
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            /*
+             * Now the adapter can resolve things from the container.
+             * For example, we could get a settings object from the container:
+             * ISettingsObj settings = _containter.GetInstance<ISettingsObj>();
+             * _timer = new Timer(this.DoWork, null, TimeSpan.Zero, TimeSpan.FromSeconds(settings.TimerInterval));
+             */
+            _timer = new Timer(this.DoWork, null, TimeSpan.Zero, TimeSpan.FromSeconds(5));
+            return Task.CompletedTask;
+        }
+
+        private void DoWork(object state)
+        {
+            // Run operation in a scope
```

Comment:
I know this comment originates from my own SO answer, but I think it can be removed, as the code line below implies this.

Reply:
Done.